src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1123,28 +1123,15 @@ internal override void CreateAsync()
                         _processInstance.RunspacePool.Dispose();
                     }
 
-                    stdInWriter = _processInstance.StdInWriter;
-                    // if (stdInWriter == null)
-                    {
-                        _serverProcess.OutputDataReceived += new DataReceivedEventHandler(OnOutputDataReceived);
-                        _serverProcess.ErrorDataReceived += new DataReceivedEventHandler(OnErrorDataReceived);
-                    }
-
                     _serverProcess.Exited += new EventHandler(OnExited);
-
-                    // serverProcess.Start();
                     _processInstance.Start();
 
-                    if (stdInWriter != null)
+                    stdInWriter = _processInstance.StdInWriter;
+                    if (stdInWriter == null)
```

Comment:
@PaulHigin do we really need this check? I kept it since it's there today, but it doesn't seem we really need it.

Reply:
I don't know ... I didn't write this code.  It looks like an attempt to protect against CreateAsync() from being called multiple times, or re-using a 'PowerShellProcessInstance' object.  Background jobs don't do this but maybe Workflow jobs did.  Probably not needed.